### PR TITLE
Persist item in DB not just in cache

### PR DESCRIPTION
**Pull request by PoLaKoSz**

### CODE REVIEW

File: src/Executables/Game/World/Entities/PlayerEntity.cs [hunk 1]
```diff
@@ -122,8 +123,8 @@ public EAntiFlags AntiFlagGender
             _world = world;
             _logger = logger;
             _scope = serviceProvider.CreateScope();
-            var itemRepository = _scope.ServiceProvider.GetRequiredService<IItemRepository>();
-            Inventory = new Inventory(itemManager, _cacheManager, _logger, itemRepository, player.Id,
+            _itemRepository = _scope.ServiceProvider.GetRequiredService<IItemRepository>();
+            Inventory = new Inventory(itemManager, _cacheManager, _logger, _itemRepository, player.Id,
```
Comment (PoLaKoSz) [inline src/Executables/Game/World/Entities/PlayerEntity.cs:127]: Didn't investigate why the ItemRepository injected to the Inventory like this.

File: src/Data/Game.Persistence/ItemRepository.cs [hunk 2]
```diff
@@ -43,4 +47,23 @@ public async Task DeletePlayerItemsAsync(uint playerId)
     {
         await _db.Items.Where(x => x.PlayerId == playerId).ExecuteDeleteAsync();
     }
+
+    public async Task SaveItemAsync(ItemInstance item)
+    {
+        _db.Items.Add(new Entities.Item()
```
Comment (MeikelLP) [inline src/Data/Game.Persistence/ItemRepository.cs:53]:
```suggestion
        _db.Items.Add(new Entities.Item
```

File: src/Data/Game.Persistence/ItemRepository.cs [hunk 3]
```diff
@@ -43,4 +47,23 @@ public async Task DeletePlayerItemsAsync(uint playerId)
...
+            Id = Guid.NewGuid(),
+            PlayerId = item.PlayerId,
+            ItemId = item.ItemId,
+            Window = item.Window,
+            Position = item.Position,
+            Count = item.Count,
+            CreatedAt = DateTime.UtcNow,
+            UpdatedAt = DateTime.UtcNow,
+        });
+        await  _db.SaveChangesAsync();
```
Comment (MeikelLP) [inline src/Data/Game.Persistence/ItemRepository.cs:64]:
```suggestion
        await _db.SaveChangesAsync();
```

File: src/Data/Game.Persistence/ItemRepository.cs [hunk 4]
```diff
@@ -43,4 +47,23 @@ public async Task DeletePlayerItemsAsync(uint playerId)
...
+
+        string key = "item:" + item.Id;
```
Comment (MeikelLP) [inline src/Data/Game.Persistence/ItemRepository.cs:66]:
```suggestion
        string key = $"item:{item.Id}";
```

File: src/Executables/Game/Commands/GiveItemCommand.cs [hunk 5]
```diff
@@ -42,16 +42,15 @@ public async Task ExecuteAsync(CommandContext<GiveCommandOptions> context)
                     return;
                 }
 
-                var instance = new ItemInstance { Id = Guid.NewGuid(), ItemId = item.Id, Count = context.Arguments.Count };
+                var instance = new ItemInstance { Id = Guid.NewGuid(), ItemId = item.Id, Count = context.Arguments.Count, PlayerId = context.Player.Player.Id };
```
Comment (MeikelLP) [inline src/Executables/Game/Commands/GiveItemCommand.cs:45]: I think the line is getting too long. Can you please add line breaks? :)

File: src/Executables/Game/Commands/ItemCommand.cs [hunk 6]
```diff
@@ -69,14 +69,14 @@ public async Task ExecuteAsync(CommandContext<ItemCommandOptions> context)
                 }
             }
 
-            var instance = new ItemInstance {Id = Guid.NewGuid(), ItemId = item.Id, Count = context.Arguments.Count};
+            var instance = new ItemInstance {Id = Guid.NewGuid(), ItemId = item.Id, Count = context.Arguments.Count, PlayerId = context.Player.Player.Id};
```
Comment (MeikelLP) [inline src/Executables/Game/Commands/ItemCommand.cs:72]: Same here?

**Review by MeikelLP — CHANGES_REQUESTED** [review]

**Comment by MeikelLP** [comment]
Did you do a rebase?
It's harder to track changes if you do so.

Please do multiple commits in the future - the PR can be squashed if you wish so

**Review by MeikelLP — APPROVED** [review]

File: src/Executables/Game/World/Entities/PlayerEntity.cs (hunk 1 above)
Comment (PoLaKoSz) [reply src/Executables/Game/World/Entities/PlayerEntity.cs:127]: Item interaction (UseItem, DropItem, etc) with this change causes EF Core exceptions.
I'm sad but i would recommend to revert this PR (i spend ~2 hours to fix it, but couldn't fully).

Comment (MeikelLP) [reply src/Executables/Game/World/Entities/PlayerEntity.cs:127]: @PoLaKoSz what are the exact errors?

**Pull request merged**